src/main/kotlin/teaforge/platform/RoboRio/internal/Subscriptions.kt

```diff
@@ -80,6 +85,20 @@ fun <TMessage, TModel> createHidPortValueState(
         )
 }
 
+fun <TMessage, TModel> createInterval(
+        model: RoboRioModel<TMessage, TModel>,
+        config: Subscription.Interval<TMessage>
+): Pair<RoboRioModel<TMessage, TModel>, SubscriptionState<TMessage>>{
+   val currentTime = HALUtil.getFPGATime()
+   return Pair(
+           model,
+           SubscriptionState.Interval(
+                   config = config,
+                   nextReadTimeMicroseconds = config.millisecondsBetweenReads * 1_000L + currentTime
```

Comment:
I would suggest always being explicit about order of operations

```suggestion
                   nextReadTimeMicroseconds = (config.millisecondsBetweenReads * 1_000L) + currentTime
```